### PR TITLE
Fix crash in explorer background context menu logic

### LINKED ISSUE #8936: Context Menu -> Open in Windows Terminal -> "The server threw an exception"
Note:
1. Right Clicking on a folder and then opening the terminal works.
2. Going inside the folder, then right clicking on empty space, and then opening the terminal gives error.
3. I believe this was a known issue with older windows versions, that is why it was disabled earlier, but i think i saw a PR that implemented the functionality differently for older windows, but it is still not working for me
4. I restarted explorer.exe, still gives the error. Maybe restarting the computer will make it work? I will try later and update the issue if it works. EDIT: Nope, still not fixed.

image

image

# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18363.1256]
Windows Terminal version: Windows Terminal Preview v1.6.10272.0

Any other software?: Not applicable
```

# Steps to reproduce

Context Menu -> Open in Windows Terminal -> "The server threw an exception"

# Expected behavior

It should work

# Actual behavior

It doesn't

### PR DESCRIPTION
**Pull request by hereafter**
Fix a bug brought in with PR: #8638 

see,
#8936 
#8638

* [x] Closes #8936
* [x] CLA signed
* [x] Tests passed

With the help from @nc-x, the issue is reproduced and fixed by this patch.

CLSCTX_IN_PROCESS is not good enough for all cases to create IShellWindows interface.
Put a CLSCTX_ALL fixes the issue.

Another debugging warning dialogs  for reusing not null com_ptr in the loop is fixed too.
(This was shown in debug builds only)

### CODE REVIEW

File: src/cascadia/ShellExtension/OpenTerminalHere.cpp [hunk 1]
```diff
@@ -266,7 +266,16 @@ std::wstring OpenTerminalHere::_GetPathFromExplorer() const
...
-    auto shell = create_instance<IShellWindows>(CLSID_ShellWindows);
+    com_ptr<IShellWindows> shell;
+    try
+    {
+        shell = create_instance<IShellWindows>(CLSID_ShellWindows, CLSCTX_ALL);
+    }
+    catch (...)
+    {
+        //look like try_create_intance is not available no more
+    }
```
Comment (zadjii-msft) [inline src/cascadia/ShellExtension/OpenTerminalHere.cpp:277]: You might be able to make this even easier with `try_create_instance`
```suggestion
    com_ptr<IShellWindows> shell{ try_create_instance<IShellWindows>(CLSID_ShellWindows, CLSCTX_ALL) };
```

**Review by zadjii-msft — APPROVED** [review]

File: src/cascadia/ShellExtension/OpenTerminalHere.cpp (hunk 1 above)
Comment (hereafter) [reply src/cascadia/ShellExtension/OpenTerminalHere.cpp:277]: it looks like try_create_instance is not is the "winrt/base.h" header. or is it?

Comment (zadjii-msft) [reply src/cascadia/ShellExtension/OpenTerminalHere.cpp:277]: You know what, we might not be on the version of cppwinrt that supports this. https://github.com/microsoft/cppwinrt/pull/663. So ignore me 😃

File: src/cascadia/ShellExtension/OpenTerminalHere.cpp [hunk 2]
```diff
@@ -266,7 +266,16 @@ std::wstring OpenTerminalHere::_GetPathFromExplorer() const
...
 
-    auto shell = create_instance<IShellWindows>(CLSID_ShellWindows);
+    com_ptr<IShellWindows> shell;
+    try
+    {
+        shell = create_instance<IShellWindows>(CLSID_ShellWindows, CLSCTX_ALL);
+    }
+    catch (...)
+    {
+        //look like try_create_intance is not available no more
```
Comment (carlos-zamora) [inline src/cascadia/ShellExtension/OpenTerminalHere.cpp:276]:
```suggestion
        //look like try_create_instance is not available no more
```
Adding this as a suggestion to make your life a bit easier :)

**Review by carlos-zamora — APPROVED** [review]

**Pull request merged**